ourownstory/test-of-time
# add split related util functions and restructure adding inputs to df

🔬 Background
We need a functionality for splitting the dataframe into train and val. Since we want to offer crossvalidation, we need a function that splits the datafram in the crossvalidation case. 
Additionally to splitting the dataframe, we want to append the last inputs of the train dataframe to the beginning of the test_data. That feature depends on the model configuration. If the model predicts based on lags or comparable histroric time steps, then the test dataframe can be added.

🔮 Key changes
Add to df_utils.py:
*`split_df()`
*`crossvalidation_split_df()`
Restructure:
* call `maybe_add_first_inputs()`, `maybe_drop_first_inputs()` within `predit()`, comment: should not be user-facing. Temporary solution.
* call `new split_df()` / `crossvalidation_split_df()` in experiment procedure. 


📋 Review Checklist
- [ ]  I have commented my code, added docstrings and data types to function definitions.
- [ ]  I have added pytests for running a single model
